src/applications/edu-benefits/0994/submitForm.js

```diff
@@ -0,0 +1,99 @@
+import _ from 'lodash';
+
+import {
+  transformForSubmit,
+  recordEvent,
+} from 'platform/forms-system/src/js/helpers';
+
+const submitFormFor = eventName =>
+  function submitForm(form, formConfig) {
+    const body = formConfig.transformForSubmit
+      ? formConfig.transformForSubmit(formConfig, form)
+      : transformForSubmit(formConfig, form);
+
+    // Copied and pasted from USFS with a couple changes:
+    // 1. Sets `withCredentials` to true
+    // 2. Sends the Authorization header with the user token
+    return new Promise((resolve, reject) => {
```

Comment:
I don't think we should copy and paste this whole block.

1. Both `withCredentials` and the auth token are included by default (see src/platform/forms-system/src/js/actions.js).
2. It looks like the main thing added in here is extra data layer info, which seems like it could be done separately from this function (my reading of the docs is that that extra info isn't associated with the event that's alongside it).

A better approach would be to update `src/platform/forms-system/src/js/actions.js` to export `submitToUrl` and call that directly in here:

```js
return submitToUrl(body, formConfig.submitUrl, formConfig.trackingPrefix)
  .then(results => {
    recordEvent({
      // all the extra variables
    });
    return results;
  });
```

Reply:
Would it matter that this would record the submission-successful event twice?

Reply:
You would leave out the `event` property in the `recordEvent` call you write with all the other data.

`recordEvent` is a little bit misnamed, it's a wrapper for `window.dataLayer`, which is an array that you can push events or data about a session to. So you can call `recordEvent` just to push variables to GTM. You don't have to have an `event` property in the object you pass as an argument.

https://developers.google.com/tag-manager/devguide

Reply:
Yes, I get that, but this solution pushes two `edu-0994--submission-successful` events for one submission.  Is that an issue?

Reply:
Sorry, didn't see that you'd already updated it. What I'm saying is that you can remove line 20.

Reply:
Yeah, I gotcha, but my issue is that including the event is in the AC.  I'll make the change so I can get it approved and then check to see if this is something that can be updated in the story.

Reply:
Gotcha. There's also an assumption in the story description that data layer variables are associated with events if you send them in the same object, which I thought was wrong, but I'm less sure after doing some research. If they have to be sent together, then we'd want to rework what I suggested to let you customize the event being sent. I don't know how to tell without having @nedierecel test it out (or tell us one way or another).

Reply:
Sounds good.  I can leave it as is, and then if you approve it, have @nedierecel test and let me know.